### ISSUE TITLE
kutty HTML element: allow child element to be prepended to the list

### ISSUE THREAD

**Issue by nikochiko** [pipalacademy/kutty#24]
Currently `html.HTMLElement` only has the `.add()` method for adding child elements, and it only allows appending to the end.
`html.HTMLElement` would be easier to use if we had the ability to also add elements at the top (or at any position).

#### For a card where header should go to the top
```python
from kutty import html

element = html.div(class_="card")
element.add(html.div(class_="card-body", _content="Hello world!"))
element.add(html.div(class_="card-header", _content="Greeting"), position=0)
```

#### For alerts with a dismiss button
This API will also allow us to build elements where one particular child has to always come at the end. For example, the Alert component with a dismiss button.

```python
from kutty import html

class Alert(html.HTMLElement):
    def __init__(self, dismissible=False):
        self._dismissible = dismissible
        if self._dismissible:
            super().add(self.get_dismiss_button())

    def add(self, element):
        if self._dismissible:
            return super().add(element, position=-1)
        else:
            return super().add(element)

     def get_dismiss_button(self):
         return html.button(class_="close", _content="&times;")
```

**Comment by anandology** [comment]
Makes sense. I would suggest make the methods `append` and `insert`, following the interface of list.

**Comment by nikochiko** [comment]
Renaming `add` to `append` might not be a consistent API because `add` should have support for adding multiple elements in one go, whereas `insert` or `append` only take one argument to add to the list.
I think a better API would be to provide a low-level `insert` method and an `add` method that can be customized by the components. The customized version of `add` need not always add to the end of the list, but to whatever index that makes sense.
For example, for an alert box with a close button as the last element, `add` should add elements before this button.

Whether we need a separate `append` or not depends on whether `element.children` is guaranteed to be correct all the time. If we can rely on `element.children` to be a list with all the elements, append would be equivalent to `element.insert(len(element.children), component)`. Because it is a low-level method, that is okay. 
If `element.children` is not reliable for all components, we would need a separate `append` that can be overloaded by components that save elements to somewhere other than `.children`.

**Comment by anandology** [comment]
How about having a concept of main-container, where append adds elements? The `get_main_container` method returns self by default and elements can override that to provide a different element. For alert, you can return something like an `alert-body` etc. The decorations like close button etc, will be outside the main container.

**Comment by nikochiko** [comment]
That makes sense. For elements like table and card too. But then that makes it hard to reason about whether `insert` should add elements inside or outside the main container, and how to add elements to the base.

**Comment by nikochiko** [comment]
I think let's come back to this after building more components, so that we build the right abstractions that will generalise?
